src/components/DateBlock/BlockWeeks/BlockWeeks.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { memo } from "react";
+
+import { getTargetKey } from "utils/getTargetKey";
+import { useBlockWeeks } from "hooks/useBlockWeeks";
+
+import {
+  StyledDateBlockItem,
+  StyledDateBlockString,
+  StyledDateBlockTooltip,
+  StyledMainBlock
+} from "./BlockWeeks.styled";
+import { type PropsBlockWeeks } from "./BlockWeeks.types";
+
+const BlockWeeks: React.FC<PropsBlockWeeks> = memo(
+  ({
+    move,
+    range,
+    handleClickRange,
+    handleMouseRange,
+    targetDay,
+    setTargetDay,
+    setViewTodo,
+    todos
+  }) => {
+    const { viewHoliday, setViewHoliday, data, mainPeriod, colorHoliday } =
+      useBlockWeeks(range, targetDay, setTargetDay);
+
+    const handleMouseOver = (
+      value: string,
+      year?: number,
+      month?: number,
+      day?: number
+    ) => {
+      setViewHoliday(value);
+      if (range) {
+        handleMouseRange(year, month, day, "over");
+      }
+    };
+
+    const handleMouseOut = (year?: number, month?: number, day?: number) => {
+      setViewHoliday("");
+      if (range) {
+        handleMouseRange(year, month, day, "out");
+      }
+    };
+
+    const handleClick = (year: number, month: number, day: number) => {
```

Comment:
что думаешь насчет того чтобы юзать карирование?
```suggestion
    const handleClick = (year: number, month: number, day: number) => () => {
```

Reply:
В принципе можно, для лучшей читаемости. Но тут функция не такая большая и думаю лучше оставить все в одном стиле написания